apps/prairielearn/src/ee/pages/instructorAiGenerateQuestion/instructorAiGenerateQuestion.ts

```diff
@@ -0,0 +1,117 @@
+import * as path from 'path';
+
+import * as express from 'express';
+import asyncHandler from 'express-async-handler';
+import fs from 'fs-extra';
+import klaw from 'klaw';
+import { OpenAI } from 'openai';
+
+import * as error from '@prairielearn/error';
+
+import { config } from '../../../lib/config.js';
+import { features } from '../../../lib/features/index.js';
+import { REPOSITORY_ROOT_PATH } from '../../../lib/paths.js';
+import { createServerJob } from '../../../lib/server-jobs.js';
+import { generateQuestion } from '../../lib/aiQuestionGeneration.js';
+import { buildContextForElementDocs } from '../../lib/context-parsers/documentation.js';
+import { buildContextForQuestion } from '../../lib/context-parsers/template-questions.js';
+import { openAiUserFromLocals, insertDocumentChunk } from '../../lib/contextEmbeddings.js';
+
+import { AIGeneratePage } from './instructorAiGenerateQuestion.html.js';
+
+const router = express.Router();
+
+async function syncContextDocuments(client: OpenAI, locals: Record<string, any>) {
```

Comment:
Let's also move this function into `ee/lib/contextEmbeddings`.

Reply:
Done in ff0b8fae364583747e838607ffc674c57e963e85.